components/concept/engine/src/main/java/mozilla/components/concept/engine/prompt/PromptRequest.kt

```diff
@@ -29,37 +29,52 @@ sealed class PromptRequest(
      * Value type that represents a request for a single choice prompt.
      * @property choices All the possible options.
      * @property onConfirm A callback indicating which option was selected.
+     * @property onDismiss A callback executed when dismissed.
      */
-    data class SingleChoice(val choices: Array<Choice>, val onConfirm: (Choice) -> Unit) : PromptRequest()
+    data class SingleChoice(
+        val choices: Array<Choice>,
+        val onConfirm: (Choice) -> Unit,
+        override val onDismiss: () -> Unit
```

Comment:
I will prefer to keep `onConfirm` parameter after `onDismiss` to avoid breaking behaviors for people that are not using named parameters  https://github.com/mozilla-mobile/android-components/pull/11637/files#r796931366